src/libostree/ostree-kernel-args.c

```diff
@@ -804,3 +804,25 @@ ostree_kernel_args_get_last_value (OstreeKernelArgs *kargs, const char *key)
   const OstreeKernelArgsEntry *e = entries->pdata[entries->len-1];
   return _ostree_kernel_args_entry_get_value (e);
 }
+
+/**
+ * ostree_kernel_args_append_if_missing:
+ * @kargs: a OstreeKernelArgs instance
+ * @arg: key or key/value pair to be added
+ *
+ * Appends @arg which is in the form of key=value pair to the hash table kargs->table
+ * and appends key to kargs->order if it is not in the hash table.
+ *
+ * Since: 2022.5
+ **/
+void
+ostree_kernel_args_append_if_missing (OstreeKernelArgs  *kargs,
+                                      const char *arg)
+{
+  char *duped = g_strdup (arg);
+
+  if (g_hash_table_contains (kargs->table, duped))
```

Comment:
- The value that has `=` like `TESTARG=TESTVALUE` will be appended again if call twice, for example the result will be like:

```
# options root=LABEL=MOO quiet ostree=/ostree/boot.1/testos/7bd091ad6955c0e7ef41bab65adb016275664b55efc0d10a0c8cb263da3c1e4e/0 ARGWITHOUTKEY TESTARG=TESTVALUE TESTARG=TESTVALUE
```

- But value like `ARGWITHOUTKEY` will not be appended twice as expected.

Reply:
Parse the key from `key=value` and call g_hash_table_contains (kargs->table, key)

Reply:
Right...I think the logic we want here is more like:

```
  g_autofree char *key = g_strdup (arg);
  const char *val = split_keyeq (key);

  // Don't insert a duplicate key.
  if (g_hash_table_contains (kargs->table, key))
    return; 

  ostree_kernel_args_append (kargs, arg);
```

or so?

Though, if the argument table already has e.g. `TESTARG=VAL1` and we say `append_if_missing (TESTARG=VAL2)` ...do we expect it to be ignored, or to add a new value, or to replace the existing one?  I guess in most cases we want it to be ignored, which is what will happen here.